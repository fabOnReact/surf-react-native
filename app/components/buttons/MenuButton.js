import React from 'react';
import Icon from 'react-native-vector-icons/Ionicons';
import { StyleSheet, TouchableOpacity } from 'react-native';

export default function MenuButton({ action }){
  return (
    <TouchableOpacity
      onPress={() => console.warn('press')}
      style={styles.container}
    >
      <Icon name="md-menu" 
        size={50} 
        color="white"
        style={styles.icon}/>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  container: {
    position: "absolute",
    top: 0,
    right: 15,
    zIndex: 3,
  },
  icon: {
    shadowColor: 'rgba(0,0,0, 1)', 
    shadowOffset: { height: 3, width: 3 },
    shadowOpacity: 1,
    shadowRadius: 3,
    elevation: 3,
    marginTop: -5,
  },
})
