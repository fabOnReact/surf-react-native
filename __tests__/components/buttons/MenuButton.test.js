import React, { Component } from 'react';
import {shallow} from 'enzyme';
import MenuButton from '../../../app/components/buttons/MenuButton';

describe('MenuButton', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
          const component = shallow(<MenuButton />)
            expect(component).toMatchSnapshot()
        });

        it('renders the Icon component', () => {
          const component = shallow(<MenuButton />)
          const target = component.find({ size: 50 })
          expect(target.length).toBe(1)
        });
    });
});
