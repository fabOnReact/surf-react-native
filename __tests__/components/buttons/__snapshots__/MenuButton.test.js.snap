// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton Rendering should match to snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "position": "absolute",
      "right": 15,
      "top": 0,
      "zIndex": 3,
    }
  }
>
  <Icon
    allowFontScaling={false}
    color="white"
    name="md-menu"
    size={50}
    style={
      Object {
        "elevation": 1,
        "marginTop": -5,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
        "textShadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "textShadowRadius": 1,
      }
    }
  />
</TouchableOpacity>
`;
